internal/pkg/pct/pct.go

```diff
@@ -331,7 +331,6 @@ func createTemplateFile(targetName string, configFile string, templateFile Puppe
 		log.Error().Msgf("Error: %v", err)
 		return err
 	}
-	defer file.Close()
```

Comment:
The `defer` pattern is a standard part of go for ensuring a cleanup activity is performed. In this case we need to close the file ever if this function panics. Can you explain why you're moving this to the function return?

Reply:
It was a go sec error "Unsafe defer call of a method returning an error"
I found what i thought was a resonable solution here: https://www.joeshaw.org/dont-defer-close-on-writable-files/

However on trying to find the error again i came across this, which might be a better solution:
https://github.com/securego/gosec/issues/512#issuecomment-675286833